src/app/components/query-explainer/query-explainer.service.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpClient, HttpErrorResponse } from "@angular/common/http";
+import { inject, Injectable } from "@angular/core";
+import { catchError, EMPTY, Observable } from "rxjs";
+import { AppConfigService } from "src/app/app-config.service";
+import { QueryExplainerResponse, VerifyQueryResponse } from "./query-explainer.types";
+import { ToastrService } from "ngx-toastr";
+
+@Injectable({
+    providedIn: "root",
+})
+export class QueryExplainerService {
+    private appConfigService = inject(AppConfigService);
+    private httpClient = inject(HttpClient);
+    private toastrService = inject(ToastrService);
+
+    public proccessQuery(query: string): Observable<QueryExplainerResponse> {
+        const url = new URL(`${this.appConfigService.apiServerHttpUrl}/query`);
+        const body = {
+            query: query,
+            dataFormat: "JsonAoA",
+            schemaFormat: "ArrowJson",
```

Comment:
Don't we have these constants in GraphQL generated client?

Reply:
Yes there are, but only as enum keys